### PR TITLE
chore: add note about partial write failures

### PR DESCRIPTION
**Pull request by jacobmarble**
Helps https://github.com/influxdata/influxdb_iox/issues/5226

Small clarification to 204 responses to write requests.

- [ ] Signed the [InfluxData CLA](https://www.influxdata.com/legal/cla/)
  ([if necessary](https://github.com/influxdata/docs-v2/blob/master/CONTRIBUTING.md#sign-the-influxdata-cla))
- [ ] Rebased/mergeable

### CODE REVIEW

File: api-docs/cloud-serverless/ref.yml [hunk 1]
```diff
@@ -14680,8 +14680,8 @@ tags:
       | &nbsp;Code&nbsp; | Status                   | Description      |
       |:-----------:|:------------------------ |:--------------------- |
       | `200`       | Success                  |                       |
-      | `201`       | Created                           | Successfully created a resource. The response body may contain details. |
-      | `204`       | No content      | The request succeeded, but InfluxDB doesn't typically return a response body for the operation. |
+      | `201`       | Created                  | Successfully created a resource. The response body may contain details. |
```
Comment (wiedld) [inline api-docs/cloud-serverless/ref.yml:14683]: 201 no longer exists in serverless?

Comment (jstirnaman) [reply api-docs/cloud-serverless/ref.yml:14683]: This table isn't specific to writes--we still have v2 endpoints in Serverless that respond with `201`, correct?

File: api-docs/cloud-serverless/ref.yml [hunk 2]
```diff
@@ -14680,8 +14680,8 @@ tags:
...
+      | `204`       | No content               | The request succeeded. For _writes_, a response body is returned if part of the request failed. See the write API spec for details. |
```
Comment (jstirnaman) [inline api-docs/cloud-serverless/ref.yml:14684]:
```suggestion
      | `204`       | No content               | The request succeeded. For _writes_ that have [rejected points](/influxdb/cloud-serverless/write-data/troubleshoot/#troubleshoot-rejected-points), the [`/write` and `/api/v2/write` endpoints](#tag/Write) return a response body with error details. |
```

Thanks for this. I chose to remove the blurb about writes, and just leave it more generic (maybe a bit of self-serving DRYness), but we can add it back.

**Review by jstirnaman — CHANGES_REQUESTED** [review]
Thanks for checking that--just a suggested edit.

File: api-docs/cloud-serverless/ref.yml (hunk 1 above)
Comment (jacobmarble) [reply api-docs/cloud-serverless/ref.yml:14683]: In the context of this change, I decided to leave 201. This section of the doc seems to be about the API generally, not the write API specifically.

Comment (jacobmarble) [reply api-docs/cloud-serverless/ref.yml:14683]: TBH, I thought about whether we do or don't return 201 anywhere, and I chose not to go searching through the IOX and IDPE repos.

**Pull request merged**